src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -515,6 +515,11 @@ int D4NFilterBucket::list(const DoutPrefixProvider* dpp, ListParams& params, int
     }
   } //d4n_write_cache_enabled = true
 
+  if (cache_request) {
```

Comment:
Only works if d4n_write_cache_enabled is true, because otherwise, even if the objects are in the cache, the `cache_results` will not be populated due to this if statement